cmd/libsnap-confine-private/tool.c

```diff
@@ -117,6 +117,22 @@ void sc_call_snap_update_ns_as_user(int snap_update_ns_fd,
 				 "XDG_RUNTIME_DIR=%s", xdg_runtime_dir);
 	}
 
+	const char *snap_real_home = getenv("SNAP_REAL_HOME");
+	char snap_real_home_env[PATH_MAX + sizeof("SNAP_REAL_HOME=")] = { 0 };
+	if (snap_real_home != NULL) {
+		sc_must_snprintf(snap_real_home_env,
+				 sizeof(snap_real_home_env),
+				 "SNAP_REAL_HOME=%s", snap_real_home);
+        }
+
+	const char *snap_uid = getenv("SNAP_UID");
```

Comment:
Passing UID was removed, but waiting for finalization before merging that through into all the parts.